src/java/org/apache/cassandra/db/TypeSizes.java

```diff
@@ -47,19 +48,7 @@ public static int sizeof(String value)
 
     public static int encodedUTF8Length(String st)
     {
-        int strlen = st.length();
-        int utflen = 0;
-        for (int i = 0; i < strlen; i++)
-        {
-            int c = st.charAt(i);
-            if ((c >= 0x0001) && (c <= 0x007F))
-                utflen++;
-            else if (c > 0x07FF)
-                utflen += 3;
-            else
-                utflen += 2;
-        }
-        return utflen;
+        return ByteBufUtil.utf8Bytes(st);
```

Comment:
our write is the Netty method, so why should our length logic be custom?

Reply:
Using the companion method from the same lib makes perfect sense. 
Maybe perf was the reason of the custom implementation?

Reply:
the history i found was that we used to write 0, then write the bytes, then go back and update the size header... we then switched to using the above method in [CASSANDRA-15410](https://issues.apache.org/jira/browse/CASSANDRA-15410)... 15410 looks like it was focused on ascii and that this specific change was just thrown in...  but couldn't tell more from the history